inst/include/RcppSimdJson/deserialize.hpp

```diff
@@ -443,16 +443,18 @@ inline simdjson::simdjson_result<simdjson::dom::element> parse(simdjson::dom::pa
                     parser, /* ... and decompress to a RawVector if so, then parse that */
                     utils::decompress(std::string(json), Rcpp::String(std::string(*file_type))));
             }
-            return parser.load(std::string(json)); /* otherwise, just `parser::load()` the file */
+            simdjson::padded_string content = simdjson::padded_string::load(std::string(json));
+            return parser.iterate(content); /* otherwise, just `parser::load()` the file */
         } else {
-            return parser.parse(std::string_view(json)); /* if not file, just parse the string */
+            simdjson::padded_string content = simdjson::padded_string(std::string_view(json));
```

Comment:
This is unsafe because the padding string will go out of scope (and be destroyed) at the return of the function, so you will have a document, but not its input string... so the document will be left pointing at data that might get reclaimed or overwritten at any time.